Templates/GraphicsGem/Template/Code/Source/${Name}Module.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Module/Module.h>
+
+namespace AZ::Render
+{
+    class ${Name}Module
+        : public AZ::Module
+    {
+    public:
+        AZ_RTTI(${Name}Module, "{${Random_Uuid}}", AZ::Module);
+        AZ_CLASS_ALLOCATOR(${Name}Module, AZ::SystemAllocator);
```

Comment:
I have made updates to the existing gem templates to move the RTTI, TypeInfo, Component Descriptor and Allocator definitions to cpp files.

This template should follow that same approach.
The reason why is to reduce compile time cost, as o3de compile time is suffering death by thousand cuts with unnecessary compiling of functions in headers and includes.
Here is a link to the those Template updates: https://github.com/o3de/o3de/commit/b16fb0aa4fd56df9aa214cbfeb43ca93e99ce8ff

The changes would invoke using the `AZ_TYPE_INFO_WITH_NAME_DECL` macro, `AZ_RTTI_NO_TYPE_INFO_DECL` and `AZ_CLASS_ALLOCATOR_DECL` macro in headers.

The `AZ_COMPONENT` macro can be replaced with the `AZ_COMPONENT_DECL` macro.

Reply:
Ok, I had seen it, though when I first used it I got some errors, likely they were due to missing include files. I'll do it that way and report back any issues